src/APU.cpp

```diff
@@ -373,26 +357,16 @@ void APU::updateChannel1() {
 		uint8_t waveRatio = squareWaveRatio[wavePatternDutyIndex];
 
 		// Update the sample produced in accordance with the wave frequency
-		channel1.sample = (channel1.waveIndex & 0x07) < waveRatio ? -1.0f : 1.0f;
+		channel1.sample = (channel1.waveIndex & 0x07) < waveRatio ? 0 : 16;
```

Comment:
I'm keeping everything an int between 0 and 16 for quicker math, and then converting to a float at the end right before adding it to my audio buffer.